src/nvim/mouse.c

```diff
@@ -303,6 +304,73 @@ int jump_to_mouse(int flags,
     mouse_past_bottom = true;
   }
 
+  // Adjust the clicked column position if there are concealed characters
+  // before the current column.  But only when it's absolutely necessary.
```

Comment:
Yeah.  Most of `syntax.c`'s functions don't accept a line/col as far as I can tell.